src/components/ToDoList.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useContext } from 'react';
+import { DispatchContext, StateContext } from './StateContext';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { FilterType } from '../types/FilterType';
+
+export const ToDoList = () => {
+  const { todos, filter } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const filteredTodos: () => Todo[] = () => {
+    switch (filter) {
+      case FilterType.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+      case FilterType.COMPLETED:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  };
+
+  const useFilteredTodos = filteredTodos();
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {/* This is a completed todo */}
+      {useFilteredTodos.map(todo => (
+        <div
+          data-cy="Todo"
+          className={classNames('todo', { completed: todo.completed })}
+          key={todo.id}
+        >
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+              checked={todo.completed}
+              onClick={() => {
+                dispatch({
+                  type: 'CHANGE_STATUS',
+                  id: todo.id,
+                });
+              }}
+            />
+          </label>
+          {todo.editted ? (
+            <form
+              onSubmit={() => {
+                dispatch({
+                  type: 'EDIT_TODO',
+                  id: todo.id,
+                });
+              }}
+            >
+              <input
+                data-cy="TodoTitleField"
+                type="text"
+                className="todo__title-field"
+                placeholder="Empty todo will be deleted"
+                value={todo.title}
+                onChange={e => {
+                  dispatch({
+                    type: 'CHANGE_TITLE',
+                    id: todo.id,
+                    changedTitle: e.target.value.toString(),
+                  });
+                }}
+                onKeyUp={event => {
+                  if (event.key === 'Escape') {
+                    dispatch({
+                      type: 'HANDLE_ESCAPE',
+                      id: todo.id,
+                    });
+                    dispatch({
+                      type: 'EDIT_TODO',
+                      id: todo.id,
+                    });
```

Comment:
dispatch two actions one after another is weird when it applies to one reducer. You can handle multiple state changes with one action.